src/lib/cfa/ntp_conf.rb

```diff
@@ -239,8 +243,22 @@ def self.new_from_augeas(augeas_entry)
       # @param [string] key
       def self.record_class(key)
         entry_type = key.gsub("[]", "")
-        record_class = "::CFA::NtpConf::#{entry_type.capitalize}Record"
-        Kernel.const_get(record_class)
+        record_class = "#{entry_type.capitalize}Record"
+        if CFA::NtpConf.constants.include?(record_class.to_sym)
+          CFA::NtpConf.const_get(record_class.to_sym)
+        else
+          # it is not a predefined record type, so use a generic misc record
+          res = Class.new(MiscRecord)
+          # and specify there its augeas type
+          res.define_method(:augeas_type) { entry_type + "[]" }
```

Comment:
defined_singleton_method ?

Reply:
no, it is not singleton method, it is instance method. But maybe something that can be changed and have it as class method and not instance one.